File: App/Frontend/frontend/lib/models/commentReport.dart
class CommentReport {
  int _id;
  String _username;
  int _commentID;
  DateTime _dateReported;
  int _reportStatus;

  CommentReport(this._username, this._commentID) {
    _dateReported = DateTime.now();
    _reportStatus = 0;
  }

  int get id => _id;
  String get username => _username;
  int get commentID => _commentID;
  DateTime get dateReported => _dateReported;
  int get reportStatus => _reportStatus;

  Map<String, dynamic> toJSON(){
    var map = Map<String, dynamic>();

    if(id != null)
      map["id"] = _id;

    map["username"] = _username;
    map["commentId"] = _commentID;
    map["dateReported"] = _dateReported.toIso8601String();
    map["reportStatus"] = _reportStatus;

    return map;
  }

  CommentReport.fromJSON(Map<String, dynamic> json) {
    this._id = json["id"];
    this._username = json["username"];
    this._commentID = json["commentId"];
    this._dateReported = DateTime.parse(json["dateReported"]);
    this._reportStatus = json["reportStatus"];
  }
}

File: App/Backend/Backend/Backend/Models/CommentReport.cs
using Backend.Models.Enums;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace Backend.Models
{
    public class CommentReport
    {
        [Key]
        public int Id { get; set; }
        public string UserName { get; set; }
        public int CommentId { get; set; }
        public DateTime DateReported { get; set; }
        [NotMapped]
        public string ReportedUserName { get; set; }
        [NotMapped]
        public int PostId { get; set; }
        [NotMapped]
        public int ReportsNumber { get; set; }
        public ReportStatus ReportStatus { get; set; }
        [NotMapped]
        public double ReportValidity { get; set; }
    }
}
